add initial text for sections Operations, Logical Resource Organization, and REST Binding (#33)
w3c/lws-protocol: spec/delete-resource.md

The [delete resource](https://w3c.github.io/lws-protocol/spec/#dfn-deletion) operation removes an existing resource (or, in some cases, an entire container) from the storage. This operation is akin to deleting a file or folder in a file system. Once completed, the target resource is no longer available for read or update, and its identifier becomes invalid (attempts to access it later should report it as not found, unless it is recreated).

**Inputs**:

* **Target identifier:** The identifier of the resource or container to delete.

> * **Optional recursive flag:** Applicable if the target is a container. By default, containers must be empty (no members) to be deleted. If the client explicitly indicates a *recursive delete* (where supported), it signals that the server should delete the container and all of its contents. This flag must be used with care to avoid accidental mass deletion.

Reviewer: ```suggestion
* **Optional recursive flag:** Applicable if the target is a container. By default, containers must be empty (no members) to be deleted. If the client explicitly indicates a *recursive delete* (where supported), it signals that the server should delete the container and all of its contents including any sub-containers (akin to `rm -r` on Unix-like OS). This flag must be used with care to avoid accidental mass deletion.
```

> 

Reviewer: There's a missing header or other break from the Inputs list above and the results that follow below. I'm guessing it should be this:
```suggestion

## Results

```

> * If the target is a **non-container resource** (a single item), the server will remove that resource from storage. This includes deleting its content and any associated metadata or auxiliary resources. For example, if there are separate metadata files (like access control lists or index entries related to that resource), those should also be cleaned up as appropriate. After a successful deletion, any subsequent read or update of that resource by clients should be treated as if the resource does not exist (since it’s gone).

Reviewer: ```suggestion
* If the target is a **non-container resource** (a single item), the server will remove that resource from storage. This includes deleting its content and any associated metadata or auxiliary resources. For example, if there are separate metadata files (like access control lists or index entries related to that resource), those should also be cleaned up as appropriate. After a successful deletion, any subsequent read or update of that resource by clients should be treated as if the resource has never existed (since it no longer does).
```

> * If the target is a **container resource** (collection), the server will by default only delete it if it has no members (no child resources). This is to prevent accidental deletion of large sets of data. If the container is not empty, the server will typically refuse the operation, signaling a conflict (the client attempted to delete a non-empty container without explicitly saying it’s okay to also delete everything inside). The client then has the responsibility to either delete the members first or use an explicit recursive delete if the protocol allows.

Reviewer: ```suggestion
* If the target is a **container resource** (collection), the server will by default only delete it if it has no members (no child resources). This is to prevent accidental deletion of large sets of data. If the container is not empty (the client attempted to delete a non-empty container without explicitly saying it’s okay to also delete everything inside), the server will typically refuse the operation, signaling a conflict. The client then has the responsibility to either start by deleting the members or use an explicit recursive delete (similar to `rm -r`), if the protocol allows.
```

> * If a **recursive delete** is explicitly requested (and supported by the server), the server will attempt to delete the target container and all resources contained within it, potentially including sub-containers and their contents (i.e., the entire sub-tree of that container). If a recursive delete operation fails, the server is not required to roll back the deletions that have already succeeded. The operation may result in a partially deleted state. The server `SHOULD` indicate in its error response which resource caused the failure, if possible. Recursive deletion is a dangerous operation, so it typically must be very explicit. Some protocols might require a special header or parameter for this. If the server does not support recursive deletion, it will reject attempts to delete non-empty containers.

Reviewer: ```suggestion
* If a **recursive delete** is explicitly requested (and supported by the server), the server will attempt to delete the target container and all resources contained within it, potentially including sub-containers and their contents (i.e., the entire sub-tree of that container). If a recursive delete operation fails partway through, the server is not required to roll back the deletions that have already succeeded; i.e., the operation may result in a partially deleted state. The server `SHOULD` indicate in its error response which resource caused the failure, if possible. Recursive deletion is a dangerous operation, so it typically must be explicit. Some protocols might require a special header or parameter for this. If the server does not support recursive deletion, it will reject attempts to delete non-empty containers.
```

> * Authorization checks: The server must verify that the client has permission to delete the resource (and, if recursive, also the contained resources). If not authorized, the server will deny the request. As with reads, if the client is entirely unauthenticated or not permitted, the server might refuse in a way that doesn’t reveal whether the resource existed at all (to outsiders).

Reviewer: ```suggestion
* Authorization checks: The server must verify that the client has permission to delete the resource (and, if recursive, each contained resource). If not authorized, the server will deny the request; if recursive, the denial might come partway through the process. As with reads, if the client is entirely unauthenticated or not permitted, the server might refuse in a way that doesn’t reveal (to outsiders) whether the resource existed at all.
```
